GoMeet/src/main/webapp/js/meeting-display.js

```diff
@@ -1,13 +1,28 @@
 /**
  * Toggles meeting event display depending on login status of user.
  */
-function toggleMeetingDisplay() {
-  LoginStatus.doGet().then((loginStatus) => {
+async function toggleMeetingDisplay() {
+  let meetingEventId = getMeetingEventId();
+  await LoginStatus.doGet(meetingEventId).then((loginStatus) => {
     let prompt; 
     let style; 
     if (loginStatus.loggedIn === 'true') {
-      style = 'block'; 
       prompt = '<a href=' + loginStatus.logoutUrl + '>Logout</a>';
+      // Check if the meeting event ID is valid (i.e. there exists a MeetingEvent entity 
+      // in Datastore with that key)
+      let meetingEventEntity = MeetingEventDAO.fetchMeetingEvent(meetingEventId); 
+      if (meetingEventEntity['status'] === undefined) { // No error response was sent 
+        style = 'block';
+        let meetingName = meetingEventEntity['meetingName'];
+        let durationMins = meetingEventEntity['durationMins'];
+        let durationHours = meetingEventEntity['durationHours']; 
+        let timeFindMethod = meetingEventEntity['timeFindMethod']; 
+        let guestList = meetingEventEntity['guestList']; 
+        let meetingTimeIds = meetingEventEntity['meetingTimeIds']; 
+        let meetingLocationIds = meetingEventEntity['meetingLocationIds'];
```

Comment:
The above seven variables are unused - is that because of the TO DO next line?

Reply:
Yes - would also need to refactor Anna's displayMeetingTimeForm to accept params such as the meetingTimeIds. This is to avoid each display meeting function having to fetch the meeting entity from datastore in order to access the data. Instead, with this implementation, the fetch is only called one and the data is passed to the functions.